modules/reactor/src/meshgenerators/HexagonMeshTrimmer.C

```diff
@@ -0,0 +1,578 @@
+                                                   side_to_move);
+    // If both nodes of that side is involved in the trimming interface
+    if (side_id_0_in && side_id_1_in)
+      // The side need to be removed from sideset
+      // The other node will be handled by other element's side
+      boundary_info.remove_side(mesh.elem_ptr(std::get<0>(slc_bdry_side_list[i])),
+                                std::get<1>(slc_bdry_side_list[i]),
+                                std::get<2>(slc_bdry_side_list[i]));
+    // Node 0 is on the removal side while node 1 is on the retaining side
+    else if (side_id_0_in && (side_node_0_remove != side_node_1_remove))
+    {
+      node_list_flag[std::distance(
+          node_list.begin(), std::find(node_list.begin(), node_list.end(), side_id_0))] = true;
+      const Point p0 = *mesh.node_ptr(side_id_0);
+      const Point p1 = *mesh.node_ptr(side_id_1);
+      if (MooseUtils::absoluteFuzzyEqual(p0(1), p1(1)))
+      {
+        node_list_point[std::distance(node_list.begin(),
+                                      std::find(node_list.begin(), node_list.end(), side_id_0))] =
+            twoLineIntersection(bdry_pars[0], bdry_pars[1], bdry_pars[2], 0.0, 1.0, -p0(1));
+      }
+      else
+      {
+        node_list_point[std::distance(node_list.begin(),
+                                      std::find(node_list.begin(), node_list.end(), side_id_0))] =
+            twoLineIntersection(bdry_pars[0],
+                                bdry_pars[1],
+                                bdry_pars[2],
+                                1.0,
+                                (p1(0) - p0(0)) / (p0(1) - p1(1)),
+                                -(p0(0) + (p1(0) - p0(0)) / (p0(1) - p1(1)) * p0(1)));
+      }
+    }
+    else if (side_id_1_in && (side_node_0_remove != side_node_1_remove))
+    {
+      node_list_flag[std::distance(
+          node_list.begin(), std::find(node_list.begin(), node_list.end(), side_id_1))] = true;
+      const Point p0 = *mesh.node_ptr(side_id_0);
+      const Point p1 = *mesh.node_ptr(side_id_1);
+      if (MooseUtils::absoluteFuzzyEqual(p0(1), p1(1)))
+      {
+        node_list_point[std::distance(node_list.begin(),
+                                      std::find(node_list.begin(), node_list.end(), side_id_1))] =
+            twoLineIntersection(bdry_pars[0], bdry_pars[1], bdry_pars[2], 0.0, 1.0, -p0(1));
+      }
+      else
+      {
+        node_list_point[std::distance(node_list.begin(),
+                                      std::find(node_list.begin(), node_list.end(), side_id_1))] =
+            twoLineIntersection(bdry_pars[0],
+                                bdry_pars[1],
+                                bdry_pars[2],
+                                1.0,
+                                (p1(0) - p0(0)) / (p0(1) - p1(1)),
+                                -(p0(0) + (p1(0) - p0(0)) / (p0(1) - p1(1)) * p0(1)));
+      }
+    }
```

Comment:
what about the case when there are 3 nodes of a QUAD in the trimmed region. This is the degenerate quad which turns into a TRI right?

document the node movement involved (all 3 right?) here please

Reply:
Here we are working with nodes of a certain `side`. So the maximum number is 2.